### ISSUE TITLE
Doesn't preserve domains

### ISSUE THREAD

**Issue by lperrin** [mysqljs/mysql#489]
**TL;DR**: I don't think you should change the code, but perhaps you should include a warning in the doc.

I'm using domains in my web app. I followed the sample in the documentation to attach each connection to a domain:

http://nodejs.org/api/domain.html#domain_explicit_binding

Problem is, node-mysql doesn't preserve domains. Because of the sequential protocol, the SQL parser always runs in the same domain and passes it to all callbacks.

The fix for me is quite simple. I just attach the domain in a closure:

```
function preserveDomain(done) {
  var d = process.domain;

  return function () {
    var args = arguments;

    d.run(function () {
      done.apply(null, args);
    });
  };
}
```

However, this was very puzzling and it took me a long time to understand what was going on. I think that the code should stay domain-agnostic, but maybe include a warning in the doc ?
